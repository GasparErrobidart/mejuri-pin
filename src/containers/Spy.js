import React  from 'react'

class Spy extends React.Component{

  constructor(props){
    super(props)
    this.wrapper = React.createRef()
    this.verifyProximity = this.verifyProximity.bind(this)
  }

  verifyProximity(event){
    const { proximityThreshold = 0 , onProximity } = this.props
    const rect = this.wrapper.current.getBoundingClientRect();
    if(
      rect.top - window.innerHeight - proximityThreshold <= 0 &&
      rect.top + rect.height + proximityThreshold >= 0
    ){
      if(onProximity) onProximity(this.wrapper)
    }
  }

  componentDidMount(){
    window.addEventListener('scroll', this.verifyProximity)
    window.addEventListener('resize', this.verifyProximity)
  }

  componentWillUnmount(){
    window.removeEventListener('scroll', this.verifyProximity)
    window.removeEventListener('resize', this.verifyProximity)
  }

  render(){
    const {children} = this.props
    return(
      <div ref={this.wrapper}>
        {children}
      </div>
    )
  }

}

export default Spy
