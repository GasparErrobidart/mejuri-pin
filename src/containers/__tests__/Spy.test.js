import '@testing-library/jest-dom'
import 'jest-styled-components'

import React         from 'react'
import { cleanup }   from '@testing-library/react'
import renderer      from 'react-test-renderer'

import Spy           from '../Spy'


afterEach(cleanup)

test('Renders "Spy" component', () => {
  const testText        = "Spy Test Text"
  const component       = renderer.create(<Spy>{testText}</Spy>);
  const tree            = component.toJSON()
  expect(tree).toMatchSnapshot();
});
