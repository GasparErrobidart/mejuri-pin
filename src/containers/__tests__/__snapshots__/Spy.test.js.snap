// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders "Spy" component 1`] = `
<div>
  Spy Test Text
</div>
`;
